setup.py

```diff
@@ -508,13 +604,105 @@ def setup_pytorch_extension() -> setuptools.Extension:
         name="transformer_engine_extensions",
         sources=sources,
         include_dirs=include_dirs,
-        # libraries=["transformer_engine"], ### TODO (tmoon) Debug linker errors
         extra_compile_args={
             "cxx": cxx_flags,
             "nvcc": nvcc_flags,
         },
+        **lib_kwargs
     )
 
+def setup_jax_extension() -> setuptools.Extension:
+    """Setup PyBind11 extension for JAX support"""
+    # Source files
+    src_dir = root_path / "transformer_engine" / "jax" / "csrc"
+    sources = [
+        src_dir / "extensions.cpp",
+        src_dir / "modules.cpp",
+        src_dir / "utils.cu",
+    ]
+
+    # Header files
+    include_dirs = [
+        CUDA_HOME / "include",
+        CUDNN_INCLUDE,
+        root_path / "transformer_engine" / "common" / "include",
+        root_path / "transformer_engine" / "jax" / "csrc",
+        root_path / "transformer_engine",
+    ]
+
+    # Compile flags
+    cxx_flags = [ "-O3" ]
+    nvcc_flags = [
+        "-O3",
+        "--forward-unknown-opts",
+    ]
+    if GLIBCXX_USECXX11_ABI:
+        # Core TE library was built with C++11 ABI (for PyTorch compat) so we need to do the same
+        # for all the framework extensions.
+        cxx_flags.append("-D_GLIBCXX_USE_CXX11_ABI=1")
+        nvcc_flags.append("-D_GLIBCXX_USE_CXX11_ABI=1")
```

Comment:
We should make sure this is independent of the default value of `GLIBCXX_USECXX11_ABI`:
```suggestion
    if GLIBCXX_USECXX11_ABI is not None:
        # Compile JAX extensions with same ABI as core library
        flag = f"-D_GLIBCXX_USE_CXX11_ABI={int(GLIBCXX_USECXX11_ABI)}"
        cxx_flags.append(flag)
        nvcc_flags.append(flag)
```